// @strapi/design-system/dist/Tooltip/Tooltip.js
import e from "react";
import o from "prop-types";
import v from "styled-components";
import { Box as x } from "../Box/Box.js";
import { Typography as E } from "../Typography/Typography.js";
import { Portal as T } from "../Portal/Portal.js";
import { useTooltipHandlers as h } from "./hooks/useTooltipHandlers.js";
import { useTooltipLayout as R } from "./hooks/useTooltipLayout.js";
import { useId as p } from "../helpers/useId.js";
import { VisuallyHidden as I } from "../VisuallyHidden/VisuallyHidden.js";
const H = v(x)`
  /* z-index exist because of its position inside Modals */
  z-index: 4;
  display: ${({ visible: t }) => t ? "revert" : "none"};
`, a = ({ children: t, label: n, description: r, delay: d, position: s, id: m, ...c }) => {
  const i = p(m), f = p(), { visible: l, ...u } = h(d), { tooltipWrapperRef: b, toggleSourceRef: y } = R(l, s), g = e.cloneElement(t, {
    tabIndex: 0,
    "aria-labelledby": n ? i : void 0,
    "aria-describedby": r ? i : void 0,
    ...u
  });
  return /* @__PURE__ */ e.createElement(e.Fragment, null, /* @__PURE__ */ e.createElement(T, null, /* @__PURE__ */ e.createElement(H, {
    id: i,
    background: "neutral900",
    hasRadius: !0,
    padding: 2,
    role: "tooltip",
    ref: b,
    visible: l,
    position: "absolute",
    ...c
  }, l && /* @__PURE__ */ e.createElement(I, {
    id: f
  }, r), /* @__PURE__ */ e.createElement(E, {
    as: "p",
    variant: "pi",
    fontWeight: "bold",
    textColor: "neutral0"
  }, n || r))), /* @__PURE__ */ e.createElement("span", {
    ref: y
  }, g));
};
a.defaultProps = {
  delay: 500,
  id: void 0,
  position: "top",
  label: void 0,
  description: void 0
};
a.propTypes = {
  children: o.node.isRequired,
  delay: o.number,
  description: o.string,
  id: o.string,
  label: o.string,
  position: o.oneOf(["top", "left", "bottom", "right"])
};
export {
  a as Tooltip
};
